diff --git a/osdep/path-win.c b/osdep/path-win.c
index 8f289ff83a..281c7dc8b8 100644
--- a/osdep/path-win.c
+++ b/osdep/path-win.c
@@ -78,6 +78,16 @@ static void path_init(void)
     char *path = mp_get_win_exe_subdir(tmp, "portable_config");
     if (path && mp_path_exists(path))
         portable_path = talloc_strdup(NULL, path);
+    else {
+        path = mp_get_win_exe_subdir(tmp, ".mpv");
+        if (path && mp_path_exists(path))
+            portable_path = talloc_strdup(NULL, path);
+        else {
+            path = mp_get_win_exe_subdir(tmp, "mpv");
+            if (path && mp_path_exists(path))
+                portable_path = talloc_strdup(NULL, path);
+        }
+    }
     talloc_free(tmp);
 }
 
