diff --git a/wscript b/wscript
index 99263b7ee5..b658957b02 100644
--- a/wscript
+++ b/wscript
@@ -292,7 +292,7 @@ iconv support use --disable-iconv.",
     }, {
         'name' : '--javascript',
         'desc' : 'Javascript (MuJS backend)',
-        'func': check_pkg_config('mujs', '>= 1.0.0'),
+        'func': check_pkg_config('mujs'),
     }, {
         'name': '--libass',
         'desc': 'SSA/ASS support',
