diff --git a/wscript b/wscript
index 55f97d2..00dd36b 100644
--- a/wscript
+++ b/wscript
@@ -334,7 +334,7 @@ iconv support use --disable-iconv.",
     }, {
         'name' : '--javascript',
         'desc' : 'Javascript (MuJS backend)',
-        'func': check_pkg_config('mujs', '>= 1.0.0'),
+        'func': check_pkg_config('mujs'),
     }, {
         'name': 'libass',
         'desc': 'SSA/ASS support',
